commit 9a752d37292bb377d3f7eb7755d78883fad5c439
Author: Jason Wang <jasowang@redhat.com>
Date:   Tue Feb 19 14:53:44 2019 +0800

    vhost: correctly check the return value of translate_desc() in log_used()
    
    [ Upstream commit 816db7663565cd23f74ed3d5c9240522e3fb0dda ]
    
    When fail, translate_desc() returns negative value, otherwise the
    number of iovs. So we should fail when the return value is negative
    instead of a blindly check against zero.
    
    Detected by CoverityScan, CID# 1442593:  Control flow issues  (DEADCODE)
    
    Fixes: cc5e71075947 ("vhost: log dirty page correctly")
    Acked-by: Michael S. Tsirkin <mst@redhat.com>
    Reported-by: Stephen Hemminger <stephen@networkplumber.org>
    Signed-off-by: Jason Wang <jasowang@redhat.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index cf82e7266397..5eaeca805c95 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -1784,7 +1784,7 @@ static int log_used(struct vhost_virtqueue *vq, u64 used_offset, u64 len)
 
 	ret = translate_desc(vq, (uintptr_t)vq->used + used_offset,
 			     len, iov, 64, VHOST_ACCESS_WO);
-	if (ret)
+	if (ret < 0)
 		return ret;
 
 	for (i = 0; i < ret; i++) {
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1453_9a752d37292bb377d3f7eb7755d78883fad5c439.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1453_9a752d37292bb377d3f7eb7755d78883fad5c439.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
